tripy/tripy/backend/compiler_api.py

```diff
@@ -67,36 +69,44 @@ def __init__(
             assert inp.shape_bounds.opt == (2, 4)
             assert inp.shape_bounds.max == (3, 4)
         """
-        # TODO (#252): Allow `shape` to be a shape tensor
         min_shape = []
         opt_shape = []
         max_shape = []
-        for elem in shape:
-            if isinstance(elem, numbers.Number):
-                elem = (elem,) * 3
-            elif isinstance(elem, Sequence):
-                if not all(isinstance(val, numbers.Number) for val in elem):
-                    raise_error(
-                        "Shape values must be numbers.",
-                        [f"Shape: {shape} contains an element: {repr(elem)} with non-numerical value(s)"],
-                    )
-                if len(elem) != 3:
+        if isinstance(shape, Shape):
```

Comment:
I don't think there's any value in allowing shape tensors to `compile`. `InputInfo` already expresses everything we want AFAICT.